### DISCUSSION TITLE
Having trouble with recorded_values()

### DISCUSSION THREAD

**Discussion by brivolbm7q** [Hugovdberg/PIconnect#628] [Q&A]
Hello,

Was trying to replicate some of the examples, but having trouble with recorded_value() function of PIData. I have gotten current_value, last_update, and recorded_values() all working as expected.

From your tutorial at:
https://piconnect.readthedocs.io/en/develop/tutorials/recorded_values.html#single-vs-multiple-values

I tried to implement this excerpt in particular:
```
with PI.PIServer() as server:
    point = server.search('*')[0]
    data = point.recorded_value('-5m', retrieval_mode=RetrievalMode.AT_OR_BEFORE)
    print(data)

```
I keep getting this error:
```
Python.Runtime.PythonException: since Python.NET 3.0 int can not be converted to Enum implicitly. Use Enum(int_value)
```
I tried modifying PI.py to avoid the int(retrieval_mode), but did not really have much luck. 

(I am on Python 3.9.7 in Windows if that matters.)

**Comment by Hugovdberg** [comment]
hmm, strange. It appears to be caused by Pythonnet 3.0. In previous versions the python enumeration had to be converted to an int to work, but apparently a nicer way is to convert using the actual `Enum` from the OSIsoft SDK. It should be fixed by replacing the `int(retrieval_mode)` with `AF.Data.AFRetrievalMode(int(retrieval_mode))`. (Perhaps the `int` could even be dropped, but I suspect Python.NET does not automatically convert the python `IntEnum` into an `int` before passing it to the .NET function)

**Reply by gykung** [reply]
Hi Hugovdberg, I encountered a similar error when moving my code from an older laptop (Spyder 4, Python 3.7) to a new computer (Spyder 5, Python 3.9). How do i overcome the problem? Where do I replace int(retrieval_mode) with AF.Data.AFRetrievalMode(int(retrieval_mode))? Thanks for your help!

My stackoverflow query:
https://stackoverflow.com/questions/73252397/python-python-net-3-0-osisoft-enum-related-error

**Reply by Hugovdberg** [reply]
The problem should be fixed by updating to version 0.10, sorry for the long time to implementing this.
